src/TabBar.tsx

```diff
@@ -291,8 +336,22 @@ export default class TabBar<T extends Route> extends React.Component<
               this.scrollView = el && el.getNode();
             }}
           >
-            {routes.map((route: T) => (
+            {routes.map((route: T, i: number) => (
               <TabBarItem
+                onLayout={({ nativeEvent: { layout } }) => {
+                  if (!this.isWidthDynamic(this.props)) return;
+
+                  this.actualTabWidths[i] = layout.width || tabWidth;
+
+                  // check if this is the last onLayout call
+                  if (!didAllOnLayoutsFire(this.actualTabWidths, routes.length))
+                    return;
+
+                  // all onLayout's have been fired
+                  this.setState({ tabWidths: [...this.actualTabWidths] });
+                  // clear
+                  this.actualTabWidths = [];
```

Comment:
I think we should leave this array as is. What I'm concerned about is that after initial measurement of all tabs, say later one tab label at index 3 updates, changing its width. Now `onLayout` will fire only for that tab, but we will never update the state because all other indexes will have holes in the `actualTabWidths` array.

If we leave this array as is, next time even if few of the tabs change their width, our condition for checking if all layouts have been measured will still be true and `setState` will fire correctly.

Worst case is, if multiple tabs are updating, it'll cause multiple setStates, but it's better than having incorrect state.